pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -430,3 +431,45 @@ func filterOutCondition(conditions []apps.StatefulSetCondition, condType apps.St
 func getStatefulSetKey(o metav1.Object) string {
 	return o.GetNamespace() + "/" + o.GetName()
 }
+
+func updatePodCondition(pod *v1.Pod, condition v1.PodCondition) {
+	for i, c := range pod.Status.Conditions {
+		if c.Type == condition.Type {
+			pod.Status.Conditions[i] = condition
+			return
+		}
+	}
+	pod.Status.Conditions = append(pod.Status.Conditions, condition)
+}
+
+func isInPlaceOnly(set *appsv1alpha1.StatefulSet) bool {
+	return set.Spec.UpdateStrategy.RollingUpdate != nil &&
+		set.Spec.UpdateStrategy.RollingUpdate.PodUpdatePolicy == appsv1alpha1.InPlaceOnlyPodUpdateStrategyType
+}
+
+// isToInPlaceUpdate check if the patches can be patched to existing pods.
+// Currently kube-apiserver just allow to modify spec.containers[*].image/spec.initContainers[*].image/spec.activeDeadlineSeconds in pod.
+func isToInPlaceUpdate(
```

Comment:
How about isToInPlaceUpdate ->getInPlaceUpdateSpec?

Reply:
Moved to ./pkg/util/inplaceupdate